From b258384a979bfcafd7f2047c0a24fd92978b0185 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Thu, 10 Jul 2014 01:09:17 -0400
Subject: [pci] hotplug: Avoid warning when _ADR not present

Message-id: <20140710010917.22146.5072.stgit@gir.stowe>
Patchwork-id: 85644
O-Subject: [RHEL7.1 PATCH 105/105] ACPI / PCI / hotplug: Avoid warning when _ADR not present
Bugzilla: 1114228
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1114228
Upstream Status: f26ca1d699e8b54a50d9faf82327d3c2072aaedd

commit f26ca1d699e8b54a50d9faf82327d3c2072aaedd
Author: Toshi Kani <toshi.kani@hp.com>
Date:   Wed Nov 27 13:33:09 2013 -0700

    ACPI / PCI / hotplug: Avoid warning when _ADR not present

    acpiphp_enumerate_slots() walks ACPI namenamespace under
    a PCI host bridge with callback register_slot().
    register_slot() evaluates _ADR for all the device objects
    and emits a warning message for any error.  Some platforms
    have _HID device objects (such as HPET and IPMI), which
    trigger unnecessary warning messages.

    This patch avoids emitting a warning message when a target
    device object does not have _ADR.

    Signed-off-by: Toshi Kani <toshi.kani@hp.com>
    Cc: 3.12+ <stable@vger.kernel.org> # 3.12+
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/pci/hotplug/acpiphp_glue.c b/drivers/pci/hotplug/acpiphp_glue.c
index 5b4e9eb..5cf43b1 100644
--- a/drivers/pci/hotplug/acpiphp_glue.c
+++ b/drivers/pci/hotplug/acpiphp_glue.c
@@ -279,7 +279,9 @@ static acpi_status register_slot(acpi_handle handle, u32 lvl, void *data,
 
  status = acpi_evaluate_integer(handle, "_ADR", NULL, &adr);
  if (ACPI_FAILURE(status)) {
-  acpi_handle_warn(handle, "can't evaluate _ADR (%#x)\n", status);
+  if (status != AE_NOT_FOUND)
+   acpi_handle_warn(handle,
+    "can't evaluate _ADR (%#x)\n", status);
   return AE_OK;
  }
 
-- 
1.7.1